File: LCC/LCCFeb/problem1.cpp
#include <iostream>

using namespace std;

int main() {

	int a, b, n;
	cin >> a;
	cin >> b;
	cin >> n;
	
	int max = a;
	int min = b;
	if (b > a) {
		max = b;
		min = a;
	}

	int count = 0;
	for (int i = 1; i <= min / 2; i++) {
		if (count == n)
			break;
		if (min % i == 0 && max % i == 0) {
			cout << i << " ";
			count++;
		}
	}

	if (count < n && max % min == 0)
		cout << min;


	return 0;
}

File: LCC/LCCFeb/problem1.py
from math import sqrt

a, b, n = map(int, input().split())
minimum = min(a, b)
maximum = max(a, b)

nums = []
count = 0

for i in range(int(sqrt(minimum)), 0, -1):
	if count == n:
		break
	if minimum % i == 0 and maximum % i == 0:
		nums.append(i)
		nums.append(int(minimum / i))
		count += 1

print(" ".join([str(x) for x in reversed(sorted(nums, reverse=True)[:n + 1])]))

